src/data/directory_content.rs

```diff
@@ -15,8 +25,14 @@ use crate::config::{FileDialogConfig, FileFilter};
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 pub struct DirectoryEntry {
     path: PathBuf,
+    size: Option<u64>,
+    last_modified: Option<SystemTime>,
+    created: Option<SystemTime>,
+    file_type: Option<String>,
```

Comment:
Could you move this to a Metadata struct?

Reply:
Isn't the `DirectoryItem` struct already a struct for the metadata?
see documentation:
```rust
/// Contains the metadata of a directory item.
///
/// This struct is mainly there so that the metadata can be loaded once and not that
/// a request has to be sent to the OS every frame using, for example, `path.is_file()`.
```

How should we describe the two structs then?

Reply:
We could document the structs something like this:
`DirectoryErntry` -> Contains information of a directory entry
`Metadata` -> Contains the filesystems metadata of the directory entry

... or something similar

Separating the metadata from the `DirectoryEntry` structure also makes accessing the metadata easier:
```
struct Metadata {
    pub size: Option<u64>,
    pub last_modified: Option<SystemTime>,
    ...
}

struct DirectoryEntry {
     metadata: Metadata,
}

impl DirectoryEntry {
    pub fn metadata(&self) -> &Metadata {
          &self.metadata
    }
}
```
This allows us to access the metadata like this: `directory_entry.metadata().size` and we don't need access methods for all the data